spinalcordtoolbox/scripts/sct_label_utils.py

```diff
@@ -232,10 +232,11 @@ def get_parser():
 # MAIN
 # ==========================================================================================
 def main(argv=None):
-    for i, arg in enumerate(argv):
-        if arg == '-create-seg' and len(argv) > i+1 and '-1,' in argv[i+1]:
-            raise DeprecationWarning("The use of '-1' for '-create-seg' has been deprecated. Please use "
-                                     "'-create-seg-mid' instead.")
+    if hasattr(argv, '__iter__'):
+        for i, arg in enumerate(argv):
+            if arg == '-create-seg' and len(argv) > i+1 and '-1,' in argv[i+1]:
+                raise DeprecationWarning("The use of '-1' for '-create-seg' has been deprecated. Please use "
+                                         "'-create-seg-mid' instead.")
```

Comment:
I usually use this for filling in default arguments:

```suggestion
    if argv is None: argv = []
    for i, arg in enumerate(argv):
        if arg == '-create-seg' and len(argv) > i+1 and '-1,' in argv[i+1]:
            raise DeprecationWarning("The use of '-1' for '-create-seg' has been deprecated. Please use "
                                     "'-create-seg-mid' instead.")
```


Ideally you would write `def main(argv=[]):`, but you need the `is None` workaround to avoid [python's mutable default arguments glitch](https://blog.finxter.com/the-mutable-default-argument-in-python/).

Reply:
Hrm... would writing `def main(argv=()):` avoid this, too?

Reply:
Ahh. After thinking about this more, I don't think we should have `None` as a default value at all. I think main should require `argv` to be passed, specifically as a list using type hinting


```diff
-  def main(argv=None):
+  def main(argv: list):
```

Reply:
Made an issue for this here: #3176.